tools/yamcs/proves_adapter.py

```
@@ -340,7 +464,24 @@ def _forward_tc_tcp(
                 f"[TC] Malformed TC frame from YAMCS ({len(tc_transfer_frame)} bytes): {exc}"
             )
             continue
-        tcp_sock.sendall(out_frame)
+        with write_lock:
+            tcp_sock.sendall(out_frame)
+
+
+def _bind_tc_socket(deployment: YamcsDeployment):
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `_bind_tc_socket`

(ANN202)